static/src/javascripts/projects/facia/modules/ui/container-show-more.js

```diff
@@ -0,0 +1,256 @@
+// @flow
+import bonzo from 'bonzo';
+import fastdom from 'fastdom';
+import qwery from 'qwery';
+import $ from 'lib/$';
+import config from 'lib/config';
+import fetchJson from 'lib/fetch-json';
+import mediator from 'lib/mediator';
+import reportError from 'lib/report-error';
+import timeout from 'lib/timeout';
+import userPrefs from 'common/modules/user-prefs';
+import groupBy from 'lodash/collections/groupBy';
+
+const HIDDEN_CLASS_NAME = 'fc-show-more--hidden';
+const VISIBLE_CLASS_NAME = 'fc-show-more--visible';
+const TEXT_HOOK = 'js-button-text';
+const PREF_NAME = 'section-states';
+const BUTTON_SPINNER_CLASS = 'collection__show-more--loading';
+const ARTICLE_ID_ATTRIBUTE = 'data-id';
+const ITEM_SELECTOR = '.js-fc-item';
+const STATE_DISPLAYED = 'displayed';
+const STATE_HIDDEN = 'hidden';
+const STATE_LOADING = 'loading';
+const REQUEST_TIMEOUT = 5000;
+
+type ButtonState =
+    | typeof STATE_DISPLAYED
+    | typeof STATE_HIDDEN
+    | typeof STATE_LOADING;
```

Comment:
That's what I did initially, but it didn't type check. Will go back and check my sanity on that.